packages/block-editor/src/filters/with-overlay-filter-strength-controls/controls.js

```diff
@@ -1,74 +1,93 @@
-			/>
-	);
-};
+const COLOR_PALETTES = generateColorPalettes(FILTERED_PALETTES);
 
-const ColorToolbar = ( props ) => {
+const OverlayFilterControls = ( props ) => {
 
-	return (
-		<Toolbar className="pixelgrade-hero-block-toolbar">
-			<Dropdown
-				position="bottom"
-				className="pixelgrade-hero-block-toolbar-dropdown"
-				contentClassName="components-nova--popover"
-				renderToggle={ ( { isOpen, onToggle } ) => (
-					<Button
-						onClick={ onToggle }
-						icon={ getIconSvg( 'invert' ) }
-						aria-expanded={ isOpen }
-						label={ __( 'Colors', '__plugin_txtd' ) }
-					/>
-				) }
-				focusOnMount={ false }
-				renderContent={ () => (
-          <OverlayControls { ...props } />
-				) }
-			/>
-		</Toolbar>
-	);
-};
+  const {
+    attributes,
+    setAttributes,
+  } = props;
 
-const Controls = ( props ) => {
+  const {
+    style,
+    overlayFilterStyle,
+    overlayFilterStrength
+  } = attributes;
 
   return (
-    <BlockControls>
-      <ColorToolbar { ...props } />
-    </BlockControls>
+      <ControlsSection label = { __( 'Overlay Filter' ) } group={__('Modules')}>
+        <ControlsTab label={ __( 'General' ) }>
+            <RadioControl
+              label={ __( 'Overlay Filter', '__plugin_txtd' ) }
+              selected={ overlayFilterStyle }
+              onChange={
+                ( nextFilterStyle ) => {
+                  setAttributes( { overlayFilterStyle: nextFilterStyle } )
+                  // We need to clear Duotone values.
+                  if ( nextFilterStyle === 'unitone' ) {
+                    setAttributes( { style: {
+                        ...style,
+                        color: {}
+                      }})
+                  }
+                }
+              }
+              options={ [
+                { label: __( 'Unitone' ), value: 'unitone' },
+                { label: __( 'Duotone' ), value: 'duotone' },
+              ] }
+            />
+
+            { overlayFilterStyle === 'duotone' && <DuotonePicker
+              colorPalette={ COLOR_PALETTES }
+              duotonePalette={ DUOTONE_PALETTES }
+              value = { style?.color?.duotone }
+              onChange={ ( newDuotone ) => {
+                const newStyle = {
+                  ...style,
+                  color: {
+                    ...style?.color,
+                    duotone: newDuotone,
+                  },
+                };
+                setAttributes( { style: newStyle } );
+              } }
+              disableCustomColors = { true }
+            />
+            }
+
+            { overlayFilterStyle === 'unitone' && <RangeControl
+              label={ __( 'Overlay Filter Strength', '__plugin_txtd' ) }
+              value={ overlayFilterStrength }
+              onChange={ ( nextOverlayFilterStrength ) => setAttributes( { overlayFilterStrength: nextOverlayFilterStrength } ) }
+              min={ 0 }
+              max={ 100 }
+              step={ 10 }
+            />
```

Comment:
Wouldn't or shouldn't this work for both `unitone` and `duotone` modes?

Reply:
Currently it's working working just for Unitone, and I think it should stay like that until Gutenberg implement transparency for Duotone, which I think it will happen really soon: https://github.com/WordPress/gutenberg/pull/34130